File: Visual C# .Net/Sample Change Profiles/Sample Connect/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

using KClmtrBase;
using KClmtrBase.KClmtrWrapper;

namespace Sample_Connect {
    public partial class Form1 : Form {
        private KClmtrWrap kClmtr;

        public Form1() {
            InitializeComponent();
            kClmtr = new KClmtrWrap();
        }

        private void button1_Click(object sender, EventArgs e) {
            kClmtr.connect(Convert.ToInt32(portNumber.Value));
            profileList.DataSource = kClmtr.CalFileList;
            profileText.Text = kClmtr.CalFileName;
            profileIDNumber.Value = kClmtr.CalFileID;
        }

        private void MeasureButton_Click(object sender, EventArgs e) {
            wMeasurement measure = kClmtr.getNextMeasurement(-1);
            if(KleinsErrorCodes.shouldStopMeasuring(measure.errorcode)) {
                kClmtr.stopMeasuring();
				MessageBox.Show(KleinsErrorCodes.errorCodesToString(measure.errorcode));
			} else {
                textBoxX.Text = measure.CIE1931_x.ToString();
                textBoxY.Text = measure.CIE1931_y.ToString();
                textBoxBigY.Text = measure.BigY.ToString();
            }
        }

        private void button2_Click(object sender, EventArgs e) {
            kClmtr.CalFileID = Convert.ToInt32(profileIDNumber.Value);
            profileText.Text = kClmtr.CalFileName;
        }
    }
}

File: Visual Basic .Net/Sample Connect/Sample Connect/Form1.vb
Imports KClmtrBase
Imports KClmtrBase.KClmtrWrapper
Public Class Form1

    Private kClmtr As New KClmtrWrap()
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        ' Need to set the port number before we can open it
        kClmtr.port = PortNumber.Value
    End Sub

    Private Sub Button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button2.Click
        ' Connect
        kClmtr.connect()
    End Sub
    Private Sub Button3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button3.Click
        ' Make sure it is connected
        If kClmtr.isPortOpen Then
            ' Grab the xyL and the error if there was one
            Dim xyL As wMeasurement = kClmtr.getNextMeasurement(1)
            ' If the code is 0 then there was no error
            If KleinsErrorCodes.shouldStopMeasuring(xyL.errorcode) Then
                kClmtr.stopMeasuring()
                MessageBox.Show(KleinsErrorCodes.errorCodesToString(xyL.errorcode))
            Else
                ' Filling in the xyL text boxes with the numbers
                TextX.Text = xyL.CIE1931_x
                TextY.Text = xyL.CIE1931_y
                TextL.Text = xyL.BigY
            End If
        Else
            MsgBox("Port is not open")
        End If
    End Sub
End Class
